lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -227,17 +227,16 @@ static StringRef getPortVerilogName(Operation *module, PortInfo port) {
 
 /// Return the verilog name of the port for the module.
 static StringRef getPortVerilogName(Operation *module, size_t portArgNum) {
+  auto htmo = cast<HWModuleLike>(module);
+  return getPortVerilogName(module, htmo.getPortList().at(portArgNum));
+}
+
+/// Return the verilog name of the port for the module.
+static StringRef getInputPortVerilogName(Operation *module, size_t portArgNum) {
   if (auto htmo = dyn_cast<HWTestModuleOp>(module))
-    return getPortVerilogName(module, htmo.getPortList().at(portArgNum));
-  // portArgNum is the index into the result of getAllModulePortInfos.
-  // Also ensure the correct index into the input/output list is computed.
-  char verilogNameAttr[] = "hw.verilogName";
-  auto portAttr = cast<HWModuleLike>(module).getAllPortAttrs();
-  if (auto portDict = cast_or_null<DictionaryAttr>(portAttr[portArgNum]))
-    if (auto updatedName = portDict.get(verilogNameAttr))
-      return updatedName.cast<StringAttr>().getValue();
-  // Get the original name of input port if no renaming.
-  return cast<HWModuleLike>(module).getPortName(portArgNum);
+    return getPortVerilogName(module, htmo.getPortList().atInput(portArgNum));
```

Comment:
Can this just access both of them via `PortList`  interface? `getPortVerilogName(module, cast<PortList>(module).getPortList().atInput(portArgNum))` or so?

Reply:
This function should be going away soon.